### PR TITLE
ScalafmtRunner: add dialectOverride

### PR DESCRIPTION
**Pull request by kitbellew**

### CODE REVIEW

File: scalafmt-tests/src/test/scala/org/scalafmt/config/ScalafmtConfigTest.scala [hunk 1]
```diff
@@ -67,4 +67,20 @@ class ScalafmtConfigTest extends FunSuite {
...
+  test("dialect override") {
+    val config1 = Config
+      .fromHoconString("""
+        |runner.dialect = scala213
+        |""".stripMargin)
+      .get
+    assert(!config1.runner.getDialect.allowToplevelTerms)
+    val config2 = Config
+      .fromHoconString("""
+        |runner.dialectOverride.allowToplevelTerms = true
```
Comment (tgodzik) [inline scalafmt-tests/src/test/scala/org/scalafmt/config/ScalafmtConfigTest.scala:79]: Is it only needed for toplevel terms ? Maybe we should have a dialect for scripts instead?

**Review by tgodzik — COMMENTED** [review]
Looks good, but I do have a question about the usability. Isn't it harder for the user to try to figure this out instead of having relevant predefined dialects?

File: scalafmt-tests/src/test/scala/org/scalafmt/config/ScalafmtConfigTest.scala (hunk 1 above)
Comment (kitbellew) [reply scalafmt-tests/src/test/scala/org/scalafmt/config/ScalafmtConfigTest.scala:79]: not at all, this is just a test to show that override works.

**Comment by kitbellew** [comment]
> Looks good, but I do have a question about the usability. Isn't it harder for the user to try to figure this out instead of having relevant predefined dialects?

while having predefined dialects works well, we have had complaints in the past when some change in scala3 was backported to Scala 2, or kind projector plugin had a new feature, and rather than us trying to keep up, this will allow people to enable features themselves.

**Review by tgodzik — APPROVED** [review]

**Pull request merged**